### PR TITLE
Port to lua is written

### PR DESCRIPTION
**Pull request by avaicode**
Just a lua binding,written because I need it.
Without tests.I hope it's of use.

### CODE REVIEW

**Comment by virusdefender** [comment]
Thanks.

I'll review it as soon as possible

File: bindings/Lua/judger.c [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            break;
+        case LUA_TSTRING:
+            *(char**)arg_refs[i] = _strdup(lua_tostring(L, -1));
+            break;
+        case LUA_TTABLE:
+            if (lua_checkstringtable(L, -1, (char**)arg_refs[i], 256 - 1, &table_size) == 0) {
+                destory_config(&config);
+                return luaL_error(L, "bad argument %s (item must be a string)", arg_mask[i].name);
+            }
+            *((char**)arg_refs[i] + table_size) = NULL;
```
Comment (avaicode) [inline bindings/Lua/judger.c]: Opps..I have not considered NIL at first.Fixed.
Seems it's better to have a `break` in the last case.

Comment (avaicode) [inline bindings/Lua/judger.c]: And if it's need,I can write tests.

Comment (virusdefender) [inline bindings/Lua/judger.c]: I think tests are importtant

Comment (avaicode) [inline bindings/Lua/judger.c]: yeah. But I think if you move all test cases into a folder,and make Python codes outside,folder 'tests' will be much cleaner and more reusable.
Is that a good suggestion?

**Comment by avaicode** [comment]
Now I have tested the code and fixed two problems according to it.It should work well.
Testcases are in my local repo,I don't want to push it because it may make your repo a mess.
Waiting for your merge.

**Comment by virusdefender** [comment]
Merged, thanks for your contribution

File: bindings/Lua/judger.c (hunk 1 above)
Comment (virusdefender) [inline bindings/Lua/judger.c]: @avaicode I have move all C test src files to `test/test_src` directory. If you want to add Lua binding tests, you can create new directory named `test/Lua` and write all your code there.

**Pull request merged**